### PR TITLE
Fix postal code format

### PR DESCRIPTION
**Pull request by Quentinchampenois**
#### What ? Why ? 

Postal code were saved as Integer in `address` field. 

#### Tasks 
- [x] Update postal code field from `Integer` to `String`
- [ ] Add migration for updating current postal codes
[EDIT] - Migration is no longer needed

### CODE REVIEW

**Review by armandfardeau — CHANGES_REQUESTED** [review]
Sorry @Quentinchampenois but I have an error when executing the migration.
Could you check, please?

Capture d’écran 2021-02-18 à 20 45 12

**Comment by Quentinchampenois** [comment]
Thanks for the review @armandfardeau, I could not reproduce your error. However, I tried to reduce the chances that it happens again.

I tried to reproduce with existing postal codes, and with a new database too.

**Review by paulinebessoles — APPROVED** [review]
Tested and approved by product! 
With @Quentinchampenois we were wondering if the migration is needed, as it was not requested by the client and may cause some issues.

File: db/migrate/20210218135141_update_postal_code_registration.rb [hunk 1]
```diff
@@ -3,18 +3,23 @@
 class UpdatePostalCodeRegistration < ActiveRecord::Migration[5.2]
   def up
     affected_users = users_with_postal_code
```
Comment (armandfardeau) [inline db/migrate/20210218135141_update_postal_code_registration.rb]: why assign value to another var? Use a memoized method or method directly.

File: db/migrate/20210218135141_update_postal_code_registration.rb [hunk 2]
```diff
@@ -3,18 +3,23 @@
...
-    updated_users = 0
 
-    affected_users&.each do |user|
-      next unless user[:address].present? && user[:address]['postal_code'].present?
+    unless affected_users.present? && affected_users.respond_to?(:each)
+      puts 'No users affected by migration, migration aborted !'
+      return
+    end
```
Comment (armandfardeau) [inline db/migrate/20210218135141_update_postal_code_registration.rb]:
```suggestion
      return if affected_users.empty
```
No need to puts empty values. Empty is sufficient.

File: db/migrate/20210218135141_update_postal_code_registration.rb [hunk 3]
```diff
@@ -3,18 +3,23 @@
...
 
-      current_postal_code = user[:address]["postal_code"]
+    updated_users = 0
+    affected_users.each do |user|
+      next unless user[:address].present?
+      next unless user[:address]['postal_code'].present?
```
Comment (armandfardeau) [inline db/migrate/20210218135141_update_postal_code_registration.rb]:
```suggestion
      next unless user.dig(:address, :postal_code).present?
```

File: db/migrate/20210218135141_update_postal_code_registration.rb [hunk 4]
```diff
@@ -35,7 +40,9 @@ def postal_code_need_update?(postal_code)
   def reformat_postal_code(postal_code)
     postal_code = postal_code.to_s if postal_code.is_a? Integer
 
-    return "0#{postal_code}" if postal_code.length == 4
+    if postal_code.length == 4 && postal_code.respond_to?(:to_s)
+      return "0#{postal_code}"
+    end
```
Comment (armandfardeau) [inline db/migrate/20210218135141_update_postal_code_registration.rb]: There is no need to overcheck for type, either postal code is an integer, then you can cast it to a string, or it's an integer, and you may cast it to a string.

Rails will run this migration into a transaction, if this fails we may fix it properly.

**Review by armandfardeau — CHANGES_REQUESTED** [review]

**Comment by Quentinchampenois** [comment]
Hi @paulinebessoles @armandfardeau , 

I have removed the migration from PR, it is ready for another review, thanks !

**Comment by paulinebessoles** [comment]
Tested and approved by product!

**Pull request merged**